core/templates/dev/head/domain/state_card/StateCardObjectFactory.ts

```diff
@@ -170,8 +170,18 @@ oppia.factory('StateCardObjectFactory', [
     };
 
     StateCard.prototype.setLastOppiaResponse = function(response) {
-      this._inputResponsePairs[
-        this._inputResponsePairs.length - 1].oppiaResponse = response;
+      // This check is added here to ensure that this._inputReponsePairs is
+      // accessed only if there is atleast one input response pair present.
+      // If a user clicks on restart from beginning option just after
```

Comment:
Add something here about "In the editor preview tab".

Reply:
Done